azerothcore/azerothcore-wotlk
# fix(CORE): C++ 11 rule of 3 compiant constructors

C++ 11 Rule of Three:
If a class requires a user-defined destructor, a user-defined copy
constructor, or a user-defined copy assignment operator, it almost
certainly requires all three.
https://en.cppreference.com/w/cpp/language/rule_of_three

Constructors added in commit remove warning during build, so as to
build successfully without warnings.

https://github.com/azerothcore/azerothcore-wotlk/issues/3021





## CHANGES PROPOSED:
-  Explicitly defaults destructor, copy constructor, or copy assignment operator, if one has been defined by user.


## ISSUES ADDRESSED:
- Closes 

Closes https://github.com/azerothcore/azerothcore-wotlk/issues/3021

## TESTS PERFORMED:
- Build performed, confirmed that no constructor warnings occur.
- Test performed on Linux (OpenSuse Tumbleweed)



## HOW TO TEST THE CHANGES:

- Build, view warnings. If no constructor warnings, test passed.
- Changes should not affect gameplay or core functionality.

## KNOWN ISSUES AND TODO LIST:

- [x] No foreseeable issues with these changes.


## Target branch(es):
- [x] Master commit 2bb868164e545d38da5c96a3ce683a5fcb3b72a6








 
## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR
